osu.Game.Rulesets.Catch/Objects/CatchHitObject.cs

```diff
@@ -16,27 +16,47 @@ public abstract class CatchHitObject : HitObject, IHasXPosition, IHasComboInform
     {
         public const float OBJECT_RADIUS = 64;
 
-        private float x;
+        // This value is after XOffset applied.
+        public readonly Bindable<float> XBindable = new Bindable<float>();
+
+        // This value is before XOffset applied.
+        private float originalX;
 
         /// <summary>
         /// The horizontal position of the fruit between 0 and <see cref="CatchPlayfield.WIDTH"/>.
         /// </summary>
         public float X
         {
-            get => x + XOffset;
-            set => x = value;
+            // TODO: I don't like this asymmetry.
```

Comment:
...and rightly so, this can't live in the current state. if you ask me, there should be an `XBindable`, an `XOffsetBindable`, and an `AggregateBindable` that sums the two into an `XWithOffsetBindable`. names are temporary, would actually advise changing to `PositionBindable` etc.

Reply:
I think we all agree this is bad, but in the interest of keeping the amount of possible breaking changes in this PR minimal, let's keep it as it is for now. Changes in this area have to be scrutinised hard with the possibility of breaking diffcalc.